[SRVCOM-1561] Add the Serverless operator resource requirements doc (#71573)
openshift/openshift-docs: modules/serverless-operator-installation-resource-requirements.adoc

> :_content-type: REFERENCE

Reviewer: Please add a “where used” list.

[id="serverless-operator-resource-requirements_{context}"]
= {ServerlessOperatorName} resource requirements

> The following sample setup might help you to estimate the minimum resource requirements for your {ServerlessOperatorName} installation. Note that your specific requirements might be significantly different, and might grow as your use of {ServerlessProductName} increases.

Author: OpenShift docs guidelines demand "help to" instead of "help" :man_shrugging:

> The following sample setup might help you to estimate the minimum resource requirements for your {ServerlessOperatorName} installation. Note that your specific requirements might be significantly different, and might grow as your use of {ServerlessProductName} increases.

Reviewer: ```suggestion
The following sample setup might help you to estimate the minimum resource requirements for your {ServerlessOperatorName} installation. Your specific requirements might be significantly different, and might grow as your use of {ServerlessProductName} increases.
```

The test suite used in the sample setup has the following parameters:

* An {ocp-product-title} cluster with:
** 10 workers (8 vCPU, 16GiB memory)
** 3 workers dedicated to Kafka
** 2 workers dedicated to Prometheus

> ** 5 workers remaining for both {ServerlessProductShortName} and test deployments

Reviewer 2: On a second thought, it might be good to keep it as that additional data point, to give the overall picture of the cluster :-/

* 89 test scenarios running in parallel, mainly focused on using the control plane. Testing scenarios typically have a Knative Service sending events through an in-memory channel, a Kafka channel, an in-memory broker, or a Kafka broker to either a Deployment or a Knative Service.

* 48 re-creation scenarios, where the testing scenario is repeatedly being deleted and re-created.

* 41 stable scenarios, where events are sent throughout the test run slowly but continuously.

> * The test setup contains, in total:

Reviewer: ```suggestion
* The test setup contains:
```
Purely a suggestion. IMO you do not need to explicitly say "in total".

** 170 Knative Services
** 20 In-Memory Channels
** 24 Kafka Channels
** 52 Subscriptions
** 42 Brokers
** 68 Triggers

> These are the minimal resource requirements for a Highly-Available (HA) setup discovered by the test suite:

Reviewer: ```suggestion
The following table details the minimal resource requirements for a Highly-Available (HA) setup discovered by the test suite:
```

> .Highly-Available setup minimal resource requirements

Reviewer: By updating the statement at line 29, I do not believe you need the label here.

[cols=3,options="header"]
|===
|Component
|RAM resources
|CPU resources

|{ServerlessOperatorName}
|1GB
|0.2 core

|Knative Serving
|5GB
|2.5 cores

|Knative Eventing
|2GB
|0.5 core

|Knative broker for Apache Kafka
|6GB
|1 core

|Sum
|14GB
|4.2 cores

|===

> These are the minimal resource requirements for a non-HA setup discovered by the test suite:

Reviewer: ```suggestion
The following table details the minimal resource requirements for a non-HA setup discovered by the test suite:
```

> .Non-HA setup minimal resource requirements

Reviewer: By updating the statement at line 60, I do not believe you need the label here.
